cmd/kubeapps-apis/proto/kubeappsapis/plugins/resources/v1alpha1/resources.proto

```diff
@@ -154,3 +164,73 @@ message CheckNamespaceExistsRequest {
 message CheckNamespaceExistsResponse {
     bool exists = 1;
 }
+
+// SecretType
+//
+// The type of secret. Currently Kubeapps itself only deals with OPAQUE
+// and docker config json secrets, but we define all so we can correctly
+// list the secret names with their types.
+// See https://kubernetes.io/docs/concepts/configuration/secret/#secret-types
+enum SecretType {
+    SECRET_TYPE_OPAQUE_UNSPECIFIED = 0;
+    SECRET_TYPE_SERVICE_ACCOUNT_TOKEN = 1;
+    SECRET_TYPE_DOCKER_CONFIG = 2;
+    SECRET_TYPE_DOCKER_CONFIG_JSON = 3;
+    SECRET_TYPE_BASIC_AUTH = 4;
+    SECRET_TYPE_SSH_AUTH = 5;
+    SECRET_TYPE_TLS = 6;
+    SECRET_TYPE_BOOTSTRAP_TOKEN = 7;
+}
+
+// CreateSecretRequest
+//
+// Request for CreateSecret
+message CreateSecretRequest {
+    // Context
+    //
+    // The context of the secret being created.
+    kubeappsapis.core.packages.v1alpha1.Context context = 1;
+
+    // Type
+    //
+    // The type of the secret. Valid values are defined by the Type enumeration.
+    SecretType type = 2;
+
+    // Name
+    //
+    // The name of the secret.
+    string name = 3;
+
+    // StringData
+    //
+    // The map of keys and values. Note that we use StringData here so that
+    // Kubernetes handles the base64 encoding of the key values for us.
+    // See https://kubernetes.io/docs/concepts/configuration/secret/#overview-of-secrets
+    map<string, string> string_data = 4;
+}
+
+// CreateSecretResponse
+//
+// Response for CreateSecret
+message CreateSecretResponse {}
+
+
+// GetSecretNamesRequest
+//
+// Request for GetSecretNames
+message GetSecretNamesRequest {
+    // Context
+    //
+    // The context for which the secret names are being fetched.
+    kubeappsapis.core.packages.v1alpha1.Context context = 1;
```

Comment:
I used the same for the SA method... but I'm wondering now whether we should keep a `resources`-plugin-specific `Context`  message to decouple it from the `packages` API.

Reply:
The resources plugin is coupled because of it's main purpose: to get the resources for an installed package. So even if we add a separate context, the module is still coupled by design for that one method?